core/arch/arm/kernel/boot.c

```diff
@@ -1625,3 +1627,89 @@ unsigned long __weak get_aslr_seed(void *fdt __unused)
 }
 #endif /*!CFG_DT*/
 #endif /*CFG_CORE_ASLR*/
+
+#if defined(CFG_CORE_SEL2_SPMC) && defined(CFG_CORE_RELOCATABLE)
+static void *get_fdt_from_boot_info(struct ffa_boot_info_header *hdr)
+{
+	struct ffa_boot_info *desc = NULL;
+	uint8_t content_fmt = 0;
+	uint8_t name_fmt = 0;
+	void *fdt = NULL;
+	int ret = 0;
+
+	if (hdr->signature != FFA_BOOT_INFO_SIGNATURE) {
+		EMSG("Bad boot info signature %#"PRIx32, hdr->signature);
+		panic();
+	}
+	if (hdr->version != FFA_BOOT_INFO_VERSION) {
+		EMSG("Bad boot info version %#"PRIx32, hdr->version);
+		panic();
+	}
+	if (hdr->desc_count != 1) {
+		EMSG("Bad boot info descriptor count %#"PRIx32,
+		     hdr->desc_count);
+		panic();
+	}
+	desc = (void *)((vaddr_t)hdr + hdr->desc_offset);
+	name_fmt = desc->flags & FFA_BOOT_INFO_FLAG_NAME_FORMAT_MASK;
+	if (name_fmt == FFA_BOOT_INFO_FLAG_NAME_FORMAT_STRING)
+		DMSG("Boot info descriptor name \"%16s\"", desc->name);
+	else if (name_fmt == FFA_BOOT_INFO_FLAG_NAME_FORMAT_UUID)
+		DMSG("Boot info descriptor UUID %pUl", (void *)desc->name);
+	else
+		DMSG("Boot info descriptor: unknown name format %u", name_fmt);
+
+	content_fmt = (desc->flags & FFA_BOOT_INFO_FLAG_CONTENT_FORMAT_MASK) >>
+		      FFA_BOOT_INFO_FLAG_CONTENT_FORMAT_SHIFT;
+	if (content_fmt != FFA_BOOT_INFO_FLAG_CONTENT_FORMAT_ADDR) {
+		EMSG("Bad boot info content format %u, expected 0 (address)",
+		     content_fmt);
+		panic();
+	}
+
+	fdt = (void *)(vaddr_t)desc->addr;
+	ret = fdt_check_full(fdt, desc->size);
+	if (ret < 0) {
+		EMSG("Invalid Device Tree at %p: error %d", fdt, ret);
+		panic();
+	}
+	return fdt;
+}
+
+static void get_sec_mem_from_manifest(void *fdt, paddr_t *base, size_t *size)
+{
+	int ret = 0;
+	uint64_t num;
+
+	ret = fdt_node_check_compatible(fdt, 0, "arm,ffa-manifest-1.0");
+	if (ret < 0) {
+		EMSG("Invalid FF-A manifest at %p: error %d", fdt, ret);
+		panic();
+	}
+	ret = dt_getprop_as_number(fdt, 0, "load-address", &num);
+	if (ret < 0) {
+		EMSG("Can't read \"load-address\" from FF-A manifest at %p: error %d",
+		     fdt, ret);
+		panic();
+	}
+	*base = num;
+	ret = dt_getprop_as_number(fdt, 0, "mem-size", &num);
```

Comment:
Where can I find specification for the FF-A manifest format that include `mem-size` subnode? On https://trustedfirmware-a.readthedocs.io/en/latest/components/ffa-manifest-binding.html, there seems to be no `mem-size`.